components/homme/src/theta-l/share/model_init_mod.F90

```diff
@@ -111,11 +111,13 @@ subroutine model_init2(elem,hybrid,deriv,hvcoord,tl,nets,nete )
          call test_imex_jacobian(elem,hybrid,hvcoord,tl,nets,nete)
 #endif
 
+    nlev_tom=0
```

Comment:
I believe this is a potential race condition. Thread 1 could lag far behind 0 (master). Thread 0 enters the `omp master` block and sets nlev_tom, then waits at the barrier. Thread 1 finally catches up and runs `nlev_tom=0`, then goes to the barrier. End result: nlev_tom is incorrectly 0.

I see no harm in having the initialization nlev_tom=0 occur within the omp master block. Or perhaps I'm missing something?

Reply:
Thanks Andrew. As nlev_tom is shared I certainly want to be more careful. I made the change.